### PR TITLE
feat: add MsBackendMetaboLights import/export

### PR DESCRIPTION
**Pull request by jorainer**
- Add save/read support for `MsBackendMetaboLights` objects.

### CODE REVIEW

**Comment by jorainer** [comment]
Need to wait for the `MsBackendMetaboLights` updates.

File: R/MsBackendMzR.R [hunk 1]
```diff
@@ -88,9 +88,14 @@ setMethod("readMsObject", signature(object = "MsBackendMzR",
...
 #' @rdname AlabasterParam
 setMethod("saveObject", "MsBackendMzR", function(x, path, ...) {
+    .save_object_spectra_data(x, path, object = "ms_backend_mz_r")
+})
+
+.save_object_spectra_data <- function(x, path, object, version = "1.0") {
     dir.create(path = path, recursive = TRUE, showWarnings = FALSE)
-    saveObjectFile(path, "ms_backend_mz_r",
-                   list(ms_backend_mz_r =list(version = "1.0")))
+    l <- list(list(version = version))
```
Comment (philouail) [inline R/MsBackendMzR.R:96]: I am a bit confused about the `list(list())`. and then you name it 'ms_backend_mzr` into it next line ?   why does it need to be a list of a list ?

File: R/PlainTextParam.R [hunk 2]
```diff
@@ -48,6 +51,11 @@
 #'   `readMsObject()` functions for other classes (such as `Spectra`,
 #'   `MsExperiment` etc).
 #'
+#' @param offline For `readMsObject()` to load MS data as a
+#'   `MsBackendMetaboLights()`: `logical(1)` to evaluate the local file cache
+#'   and only load local files. Thus `offline = TRUE` does not need an active
```
Comment (philouail) [inline R/PlainTextParam.R:56]: I would avoid repetition maybe and instead maybe `..to evaluate **only** the local file cache`  ?

**Review by philouail — APPROVED** [review]
I have small comments but otherwise everything is good for me ! thanks a lot

File: R/MsBackendMzR.R (hunk 1 above)
Comment (jorainer) [reply R/MsBackendMzR.R:96]: This is the particular format `saveObjectFile` want the input parameter, i.e. a `list` with all optional parameters (one of them being `version` and the name of that `list` should be the name/type of the object. It's confusing, I agree, but I don't know how to call it differently, given that we want to support setting the type with parameter `object` of the upstream function.

File: R/PlainTextParam.R (hunk 2 above)
Comment (jorainer) [reply R/PlainTextParam.R:56]: Good point, thanks!

Comment (jorainer) [reply R/PlainTextParam.R]: Done with the latest commit.

**Comment by jorainer** [comment]
Thanks for the review Phili!

**Pull request merged**